dali/operators/reader/file_reader_op.cc

```diff
@@ -44,6 +44,15 @@ When traversing subdirectories, the labels are assigned consecutive numbers.)cod
 
 ``stick_to_shard`` and ``random_shuffle`` cannot be used when this argument is set to True.)code",
       false)
+  .AddOptionalArg<vector<string>>("files", R"(A list of file paths to read the data from.
+
+If ``file_root`` is provided, it is prepended to the file paths.).
+When using ``files``, the labels are taken from ``labels`` argument or, if it was not supplied,
+contain indices at which given file appeared in the ``files`` list.)", nullptr)
+  .AddOptionalArg<vector<int>>("labels", R"(Labels acoompanying contents of files listed in
```

Comment:
I wonder about providing the zipped list of either `[files]` or [(files, labels)]` and ditching the `labels` argument.

Reply:
That's a great idea, but not quite doable - there's no `DALI_STRING_INT_PAIR_VEC` type.